### PR TITLE
feat: add estimatePrecursorMz function (issue #315)

### PR DESCRIPTION
**Pull request by jorainer**
- Add the `estimatePrecursorMz` functions to estimate/adjust the reported precursor m/z for DDA spectra.

This PR fixes issue #315 , i.e. for DDA data from Waters instruments, the reported precursor m/z value might not be correct because it is not being corrected through the lockmass calibration. Would also be good to get hold of some more (different) Waters DDA files to check.

### CODE REVIEW

File: R/Spectra-estimatePrecursorMz.R [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#' spectrum, this method identifies the peak with the highest intensity and
+#' an m/z value similar to the fragment spectrum's reported precursor m/z
+#' (given parameters `tolerance` and `ppm`). This m/z value is then reported.
+#' Since the fragment spectrum's potential MS1 mass peak is selected based on
+#' its intensity, this method should **only be used for DDA data**.
+#'
+#' @note
+#'
+#' This approach should only be applied if fragment spectra are recorded
+#' through  data dependent acquisition (DDA), since only with this data it
```
Comment (philouail) [inline R/Spectra-estimatePrecursorMz.R:26]: there's an extra space between "through" and "data".

Comment (philouail) [reply R/Spectra-estimatePrecursorMz.R:26]: Also I'm a bit confused by the sentence, maybe rephrasing would help: 

"This approach is applicable only when fragment spectra are obtained through data-dependent acquisition (DDA), as it assumes that the peak with the highest intensity within the given m/z window corresponds to the precursor ion."

**Review by philouail — APPROVED** [review]
All good to me, super cool function and good that this issue was discovered !  I was just confused by one paragraph in the documentation, I put an alternative in case.

File: R/Spectra-estimatePrecursorMz.R (hunk 1 above)
Comment (jorainer) [reply R/Spectra-estimatePrecursorMz.R:26]: Oh yes, great, thanks! that's much better. I'll fix.

**Comment by jorainer** [comment]
@mar-garcia , can you please check the documentation of the function? just to ensure it's correct and describes the problem properly.

**Comment by mar-garcia** [comment]
Not sure how I can check the documentation.... I've just checked what is written here: https://github.com/rformassspectrometry/Spectra/blob/adjust_precursor_mz/R/Spectra-estimatePrecursorMz.R
The only small detail I would add the specifcation "in the previous MS1 spectrum" in the following sentence:
Thus, for each fragment spectrum, this method identifies **_in the previous MS1 spectrum_** the peak with the highest intensity and an m/z value similar to the fragment spectrum's reported precursor m/z (given parameters `tolerance` and `ppm`).

**Comment by jorainer** [comment]
that's perfect, thanks @mar-garcia !

**Pull request merged**